custom-update-agent/updateagent/update_manager.go

```diff
@@ -0,0 +1,192 @@
+		return
+	}
+	if operation.GetActivityID() != activityID {
+		log.Warn("Ignoring received command %s for baseline %s and activityId %s, but not matching operation in progress [%s].",
+			command.Command, command.Baseline, activityID, operation.GetActivityID())
+		return
+	}
+	operation.Execute(command.Command, command.Baseline)
+}
+
+// Get returns the current state as an inventory graph.
+// The inventory graph includes a root software node (type APPLICATION) representing the update agent itself and a list of software nodes (type DATA) representing the available files.
+func (updMgr *fileUpdateManager) Get(ctx context.Context, activityID string) (*types.Inventory, error) {
+	return toInventory(updMgr.asSoftwareNode(), updMgr.getCurrentFiles()), nil
+}
+
+func toInventory(swNodeAgent *types.SoftwareNode, swNodeFiles []*types.SoftwareNode) *types.Inventory {
+	swNodes := []*types.SoftwareNode{swNodeAgent}
+	associations := []*types.Association{}
+	if len(swNodeFiles) > 0 {
+		swNodes = append(swNodes, swNodeFiles...)
+		for _, swNodeContainer := range swNodeFiles {
+			swNodeContainer.ID = swNodeAgent.Parameters[0].Value + ":" + swNodeContainer.ID
+
+			associations = append(associations, &types.Association{
+				SourceID: swNodeAgent.ID,
+				TargetID: swNodeContainer.ID,
+			})
+		}
+	}
+	return &types.Inventory{
+		SoftwareNodes: swNodes,
+		Associations:  associations,
+	}
+}
+
+func (updMgr *fileUpdateManager) asSoftwareNode() *types.SoftwareNode {
+	return &types.SoftwareNode{
+		InventoryNode: types.InventoryNode{
+			ID:      updMgr.Name() + "-update-agent",
+			Version: version.ProjectVersion,
+			Name:    updateManagerName,
+			Parameters: []*types.KeyValuePair{
+				{
+					Key:   parameterDomain,
+					Value: updMgr.Name(),
+				},
+			},
+		},
+		Type: types.SoftwareTypeApplication,
+	}
+}
+
+func (updMgr *fileUpdateManager) getCurrentFiles() []*types.SoftwareNode {
+	files := []*util.File{}
+	propsFile, err := os.Open("./fileagent/state.props")
+	if err != nil {
+
+		slog.Error("got error when checking current files", "error", err)
+		return nil
+
+	}
+
+	properties, err := props.Read(propsFile)
+	if err != nil {
+
+		slog.Error("got error when reading state.props file", "error", err)
+		return nil
+
+
+	} else {
+		for _, filename := range properties.Names() {
+			url, _ := properties.Get(filename)
+			files = append(files, &util.File{Name: filename, DownloadURL: url})
+		}
+	}
+	return util.FromFiles(files)
+}
+
+// Dispose releases all resources used by this instance
+func (updMgr *fileUpdateManager) Dispose() error {
+	return nil
+}
+
+// WatchEvents subscribes for events that update the current state inventory
+func (updMgr *fileUpdateManager) WatchEvents(ctx context.Context) {
+	// no container events handled yet - current state inventory reported only on initial start or explicit get request
```

Comment:
Copy-paste? I think you can add just some description what should be implemented in that function.